resdk/resources/relation.py

```diff
@@ -71,21 +67,21 @@ def samples(self):
     @property
     def collection(self):
         """Return collection object to which relation belongs."""
-        if not self._hydrated_collection:
-            self._hydrated_collection = self.resolwe.collection.get(self._collection)
-        return self._hydrated_collection
+        if not self._collection:
+            self._collection = self.resolwe.collection.get(
+                self._original_values.get('colection', None)
+            )
+        return self._collection
 
     @collection.setter
-    def collection(self, collection):
+    def collection(self, payload):
         """Set collection to which relation belongs."""
-        self._collection = get_collection_id(collection)
-        # Save collection if already hydrated, othervise it will be rerived in getter
-        self._hydrated_collection = collection if is_collection(collection) else None
+        self._resource_setter(payload, Collection, '_collection')
 
     def update(self):
         """Clear cache and update resource fields from the server."""
         self._samples = None
-        self._hydrated_collection = None
+        self._collection = None
```

Comment:
You no longer need to clear cache for these objects as they are not lazy loaded anymore.

Reply:
But they ares still cached, so it makes sense to clear cache at update.

Reply:
If related object is included in object it will automatically update it. Setting it to none only makes sense if related object is retrieved in a separate request and you mark that it needs to be done again.

Reply:
Agh, sorry, you/re right. Removed.